public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -207,6 +214,32 @@ export const GeneratePopoverBody: React.FC<{
     return generateInsight();
   };
 
+  useEffect(() => {
+    if (insightAvailable && showInsight && insight === '') {
+      const { contextContent, dataSourceQuery, summaryType, insightType } = getInsightParams(
+        contextObject,
+        incontextInsight
+      );
+
+      onGenerateInsightBasedOnSummary(
```

Comment:
I think we should call `onGenerateInsightBasedOnSummary` when `View insights` button is clicked, the component should ideally have a insight loading state, so when switching between summary/insights back and forth, the insight API won't be called many times.